### PR TITLE
Refactor of ammo and it's children file to clean a bit BE code.

### PR DESCRIPTION
**Pull request by Huffie56**
# About the pull request
Make it so that all child of ammo can use BE code instead of just bullet child.

# Explain why it's good for the game
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
refactor: moved the two proc that handled BE to ammo object instead of bullet object allowing more gun to BE in the future.
refactor: removed some instance where variable where change for the value they already had.
/:cl:

### CODE REVIEW

**Comment by Huffie56** [comment]
noticed that damage isn't set by define value ... not sure what to do with this but i could refactor this here if needed...

File: code/datums/ammo/bullet/pistol.dm [hunk 1]
```diff
@@ -167,8 +166,6 @@
 
 /datum/ammo/bullet/pistol/highpower
 	name = "high-powered pistol bullet"
-	headshot_state = HEADSHOT_OVERLAY_MEDIUM
```
Comment (fira) [inline code/datums/ammo/bullet/pistol.dm:170]: why remove the overlays?

Comment (Huffie56) [reply code/datums/ammo/bullet/pistol.dm:170]: because it is already set the same way  in is parent /datum/ammo/bullet/pistol

**Comment by Huffie56** [comment]
https://discord.com/channels/150315577943130112/964684928161808384/1175457155604828251

**Comment by Birdtalon** [comment]
Lets say at some point in the future we want to allow BE with other types of ammo. The user feedback and applied effects of these procs don't fit all `/datum/ammo`  so it would have to be refactored at some point. When are we going to see BE with `/datum/ammo/xeno/acid` 😄. But to be honest I'm not sure I see the benefit of moving these out of the scope of `datum/ammo/bullet` at this time.

**Comment by Huffie56** [comment]
> Lets say at some point in the future we want to allow BE with other types of ammo. The user feedback and applied effects of these procs don't fit all `/datum/ammo` so it would have to be refactored at some point. When are we going to see BE with `/datum/ammo/xeno/acid` 😄. But to be honest I'm not sure I see the benefit of moving these out of the scope of `datum/ammo/bullet` at this time.

yea i am probably going to close this PR and do something else around those file their is lot's to do on them....
turning random damage into define come to mind...

**Comment by Huffie56** [comment]
not a real point to this PR closing it down

**Pull request closed without merging**